### PR TITLE
avoid passing variables in JS

### PR DESCRIPTION
**Pull request by 3nids**
it is not working anymore with Qt 5.14+

in this case the QgsPointSequence was transformed to a vector, but when sent back to cpp, it's not seen as it should/would/could

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-887-711bd273a25cba06c359249b55f857585fc6b6fe-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-887-711bd273a25cba06c359249b55f857585fc6b6fe-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-887-711bd273a25cba06c359249b55f857585fc6b6fe-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-887-711bd273a25cba06c359249b55f857585fc6b6fe-x86_64.apk)

File: src/core/utils/geometryutils.cpp [hunk 1]
```diff
@@ -44,3 +44,9 @@ QgsGeometry::OperationResult GeometryUtils::addRingFromRubberBand(QgsVectorLayer
   return layer->addRing(ring, &fid);
 }
 
+QgsGeometry::OperationResult GeometryUtils::splitFeatureFromRubberBand(QgsVectorLayer *layer, RubberbandModel *rubberBandModel)
```
Comment (m-kuhn) [inline src/core/utils/geometryutils.cpp]: Can you activate your pre-commit hook ;)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-887-1fb8e6dda58a83971abca862d5e312cc03478a5c-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-887-1fb8e6dda58a83971abca862d5e312cc03478a5c-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-887-1fb8e6dda58a83971abca862d5e312cc03478a5c-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-887-1fb8e6dda58a83971abca862d5e312cc03478a5c-x86_64.apk)

File: src/qml/geometry_editors/SplitFeatureToolbar.qml [hunk 2]
```diff
@@ -23,13 +23,23 @@ VisibilityFadingRow {
...
+      {
+        displayToast( qsTr( 'Feature could not be split' ) );
+        featureModel.currentLayer.rollBack()
+        cancel()
+        finished()
+      }
+      else
+      {
+        featureModel.currentLayer.commitChanges()
+        cancel()
```
Comment (m-kuhn) [inline src/qml/geometry_editors/SplitFeatureToolbar.qml]: It seems weird to "cancel" after a successful operation.
Should this be called "closeToolbar". And ideally be directly connected to finished()?

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-887-70d11a1dc812e4c6336e3634db4cab3d2f299457-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-887-70d11a1dc812e4c6336e3634db4cab3d2f299457-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-887-70d11a1dc812e4c6336e3634db4cab3d2f299457-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-887-70d11a1dc812e4c6336e3634db4cab3d2f299457-x86_64.apk)

**Pull request merged**